ios/testmanagerd/xctestrunutils.go

```diff
@@ -141,15 +164,46 @@ func parseVersion1(content []byte) (schemeData, error) {
 		schemeBuf := new(bytes.Buffer)
 		encoder := plist.NewEncoder(schemeBuf)
 		if err := encoder.Encode(schemeMap); err != nil {
-			return schemeData{}, fmt.Errorf("failed to encode scheme %s: %w", key, err)
+			return []schemeData{}, fmt.Errorf("failed to encode scheme %s: %w", key, err)
 		}
 
 		// Decode the plist buffer into schemeData
 		decoder := plist.NewDecoder(bytes.NewReader(schemeBuf.Bytes()))
 		if err := decoder.Decode(&schemeParsed); err != nil {
-			return schemeData{}, fmt.Errorf("failed to decode scheme %s: %w", key, err)
+			return []schemeData{}, fmt.Errorf("failed to decode scheme %s: %w", key, err)
+		}
+		return []schemeData{schemeParsed}, nil
+	}
+	return []schemeData{}, nil
+}
+
+func parseXCTestRunFileFormatVersion2(content []byte) ([]schemeData, error) {
+	var testConfigs xCTestRunVersion2
+	if _, err := plist.Unmarshal(content, &testConfigs); err != nil {
+		return []schemeData{}, fmt.Errorf("failed to parse format version: %w", err)
+	}
+
+	// Check if TestConfigurations is empty
+	if len(testConfigs.TestConfigurations) == 0 {
+		return []schemeData{}, fmt.Errorf("no TestConfigurations found in XCTestRun file(format version: 2); cannot proceed with test parsing")
+	}
+
+	// Check if TestConfigurations contains more than one item
+	if len(testConfigs.TestConfigurations) > 1 {
+		return []schemeData{}, fmt.Errorf("expected exactly one TestConfiguration in XCTestRun file(format version: 2), but found %d", len(testConfigs.TestConfigurations))
+	}
```

Comment:
```suggestion
	if len(testConfigs.TestConfigurations) != 1 {
		return []schemeData{}, fmt.Errorf("The .xctestrun file you provided contained %d entries in the TestConfiguration list. This list should contain exactly 1 entry. Please revisit your test configuration so that it only contains one entry.", len(testConfigs.TestConfigurations))
	}
```